__tests__/app.test.js

```diff
@@ -139,48 +138,78 @@ describe('GET /api/:article_id/comments', () => {
-      expect(msg).toEqual("No comments found");
+      expect(msg).toBe("No comments found");
     });
   });
   test("400: responds with 'Bad request' if passed an invalid article ID", () => {
     return request(app)
       .get("/api/articles/not-an-id/comments")
       .expect(400)
       .then(({ body: { msg } }) => {
-        expect(msg).toEqual("Bad request");
+        expect(msg).toBe("Bad request: Invalid input");
       });
   });
 });
 
   //TASK 7
-describe.skip('POST /api/:article_id/comments', () => {
-  test('201: adds a comment to the given article_id', () => {
+describe('POST /api/:article_id/comments', () => {
+  test('201: responds with the posted comment', () => {
+    const newComment = { username: "butter_bridge", body: "This is a good article name"};
     return request(app)
-    .get('/api/articles/2/comments')
+    .post('/api/articles/2/comments')
+    .send(newComment)
     .expect(201)
-    .then(({body}) => {
-      const { comments } = body;
-      expect(Array.isArray(comments)).toBe(true);
-      comments.forEach((comment) => {
-        expect(comment).toEqual(
-          expect.objectContaining({
-            comment_id: expect.any(Number),
-            votes: expect.any(Number),
-            created_at: expect.any(String),
-            author: expect.any(String),
-            body: expect.any(String),
-            article_id: 3
-          })
-        );
-      });
-      });
+    .then(({ body }) => {
+      expect(body.comment).toEqual(
+        expect.objectContaining({
+          "comment_id": expect.any(Number),
+          "votes": 0,
+          "created_at": expect.any(String),
+          "author": "butter_bridge",
+          "body": "This is a good article name",
+          "article_id": 2
+        })
+      );
+    });
+    });
+  test('400: responds with error message if body or username not provided', () => {
+    const invalidComment = { body: "Missing username"};
+    return request(app)
+    .post('/api/articles/2/comments')
+    .send(invalidComment)
+    .expect(400)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Bad request: Missing required fields");
+    });
+  });
+  test('400: responds with error message if article_id is invalid', () => {
+    const newComment = { username: "butter_bridge", body: "This is a good article name"};
+    return request(app)
+    .post('/api/articles/not-a-number/comments')
+    .send(newComment)
+    .expect(400)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Bad request: Invalid input");
     });
-  });
\ No newline at end of file
+  });
+  test('404: responds with error message if article_id does not exist', () => {
+    const newComment = { username: "butter_bridge", body: "This is a good article name"};
+    return request(app)
+    .post('/api/articles/9999/comments')
+    .send(newComment)
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Article or user not found");
+    });
+  });
+});
```

Comment:
Excellent error handling, but there is one missing:

What should happen when given a username but one that does not exist?
E.g `{ username: "IDontExist", body: "something something" }`